incubator/node/README.md

```diff
@@ -83,3 +95,58 @@ The [Bitnami Node](https://github.com/bitnami/bitnami-docker-node) image stores
 
 Persistent Volume Claims are used to keep the data across deployments. This is known to work in GCE, AWS, and minikube.
 See the [Configuration](#configuration) section to configure the PVC or to disable persistence.
+
+## Set an Ingress
+
+First install the nginx-ingress controller via helm:
+
+```
+$ helm install stable/nginx-ingress
+```
+
+Now deploy the node helm chart:
+
+```
+$ helm install --name my-release bitnami-incubator/node --set ingress.enabled=true,ingress.host=example.com,serviceType=ClusterIP
+```
+
+### Configure TLS termination for your ingress controller
+
+You must manually create a secret containing the certificate and key for your domain. You can do it with this command:
+
+```
+$ kubectl create secret tls my-tls-secret --cert=path/to/file.cert --key=path/to/file.key
+```
+
+Then ensure you deploy the Helm chart with the following ingress configuration:
+
+```
+ingress:
+  enabled: false
+  path: /
+  host: example.com
+  annotations:
+    kubernetes.io/ingress.class: nginx
+  tls:
+      hosts:
+        - example.com
+```
+
+## Provision a database using the Open Service Broker for Azure
+
+1. Install Service Catalog in your Kubernetes cluster following [this instructions](https://kubernetes.io/docs/tasks/service-catalog/install-service-catalog-using-helm/)
+2. Install the Open Service Broker for Azure in your Kubernetes cluster following [this instructions](https://github.com/Azure/helm-charts/tree/master/open-service-broker-azure)
+3. Deploy the helm chart:
+
+    ```
+    $ helm install --name node-app --set externalDatabase.azure.enabled=true bitnami-incubator/node
+    ```
+
+Setting `externalDatabase.azure.enabled` to `true` makes the chart to create a `ServiceInstance` and a `ServiceBinding` kubernetes resources.
```

Comment:
Mmm, is this what we wanted to do? I thought the idea was to make it more generic (not only for Azure), but have instructions on how to use the sevice catalog to create an instance in Azure, and connect it using the generic externalDatabase values.

Reply:
We thought to do it in this way because the ServiceInstance and ServiceBinding resources are automatically created for the use. Make it more generic means that the user will need to write and create these resources on its own before deploys the chart.  
Also, the idea was to consume the database credentials from the Secret instead of passing them as standard helm values.

Reply:
So I think what we should do here is consume the secret, and not create the ServiceInstance and ServiceBinding. We can assume that those have been created, out of band, and just take in the result of that binding (i.e. the secret).

The main reason for this is that ServiceInstances usually require a range of parameters that are very specific to the cloud platform. Whilst we could easily make those parameters configurable via chart values, I think it's better to not create a direct dependency on the ServiceInstance.

This means that another scenario we can support is when you already have a ServiceInstance (and maybe also a ServiceBinding) and want to use that instead of have the chart provision one for you. With the current approach, it's not possible to use an existing ServiceInstance.

We should document the steps needed to create the ServiceInstance and ServiceBinding, either by having example YAML files or using the svcat CLI. The chart should then take in a secret (`--set externalDatabase.secretName=<name>`) and we should document what keys we expect that secret to have (for now, it should be whatever the OSBA ServiceBinding creates for us).

WDYT?